runtime/lua/vim/lsp/util.lua

```diff
@@ -1272,6 +1272,23 @@ function M.character_offset(buf, row, col)
   return str_utfindex(line, col)
 end
 
+function M.format_formatexpr()
+ -- only reformat on explicit gq command
+ -- `formatexpr` is also called when exceding
+ -- `textwidth` in insert mode
+ if vim.fn.mode() == 'i' or vim.fn.mode() == 'R' then
+    -- fall back to Vims internal reformatting
+    return 1
+ end
+ local start_line = vim.v.lnum
+ local end_line = start_line + vim.v.count - 1
+ if start_line > 0 and end_line > 0 then
+   vim.lsp.buf.range_formatting({}, {start_line, 0}, {end_line, 0})
```

Comment:
I don't think this is synchronous, right? We should make it synchronous if it's going to run in an autocmd I think.